tests/post/steps/test_liveness.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+from pytest_bdd import scenario, given, when, then, parsers
+
+# Scenarios
+@scenario('../features/pods_alive.feature', 'get pods')
+def test_pods(host):
+    pass
+
+# Given
+
+@given('The bootstrap phase is executed')
+def run_bootstrap(host):
+    pass
+
+# When
+
+@when(parsers.parse("I run '{cmd}'"))
+def run_command(request, host, cmd):
+    cmd_res = host.run(command=cmd)
+    request.cmd_result = cmd_res.stdout.strip()
+
+# Then
+
+@then('Pods list should not be empty')
+def list_pods(request):
+    assert len(request.cmd_result)
```

Comment:
I don't like how there's an implicit dependency between this `then` and the `run_command` `when`.

Reply:
isnt it defined in the feature file ?

Reply:
I think in here we should skip the `when` statement since there is no real action.

The test should be:
```
         Given The 'kubelet' service is up
         Then Pods list should not be empty in the kube-system namespace
```

The `then` statement should then build the proper command based on this and check whether the state is what we expect or not. Hence the dependency between both statements just vanishes.